### DISCUSSION TITLE
Where can I find proper troubleshooting steps/tools to identify why winget will not run without error?

### DISCUSSION THREAD

**Discussion by KirkMunroSagent** [microsoft/winget-cli#2328] [Q&A] [answered]
I love winget. I used it for close to a year, keeping all of my installed packages current weekly. It's exactly what I think needs to be in Windows for package management.

At least this was the case until it just stopped working.

Now, I'm using Windows 10 Enterprise in a locked down environment where I have a non-admin user account and an admin account that cannot log in interactively. It was difficult to get this installed manually in this environment, but I succeeded, but that was a year ago. A few months back I started getting this when I run winget:

> ResourceUnavailable: Program 'winget.exe' failed to run: An error occurred trying to start process 'C:\Windows\system32\winget.exe' with working directory 'C:\'. The file cannot be accessed by the system.At line:1 char:1
> + winget
> + \~\~\~\~\~\~.

I've looked online and found some discussion about it (in this repo I think), but ultimately nobody was taking ownership and either (a) providing steps to diagnose and fix the problem, or (b) acknowledging the problem, and identifying how it will be fixed, or starting discussions to address the problem.

From my perspective the biggest downfall to winget is the packaging of winget. Seems ironic, but I really wish that winget was more like PowerShell, with multiple installation paths, including an actual installer for installation in environments where you do not have a store, to completely remove any dependency on that from the CLI itself.

Anyway, I wanted to post this because I'm still very frustrated that I cannot get winget working (I try again from time to time, and always come up empty). Winget not working hasn't been getting in the way of my work until now, so it seems like I'm going to need to flip to chocolatey, even if I really, really don't want to start using that tool.

If anyone has some solid steps that can help identify how to troubleshoot why winget might fail to run as shown above, or how to get it working, I would appreciate it.

Thanks for listening.

**Comment by florelis** [comment]
Have you looked at the [Troubleshooting page](https://github.com/microsoft/winget-cli/blob/master/doc/troubleshooting/README.md)?

**Comment by denelon** [comment]
@KirkMunroSagent

Are you able to install by executing the .msixbundle from the [GitHub releases page](https://github.com/microsoft/winget-cli/releases)?

**Comment by OfficialEsco** [comment]
> Now, I'm using Windows 10 Enterprise in a locked down environment where I have a non-admin user account and an admin account that cannot log in interactively. It was difficult to get this installed manually in this environment, but I succeeded, but that was a year ago.

WinGet can be installed in 2 simple ways
Via the Microsoft Store https://apps.microsoft.com/store/detail/app-installer/9NBLGGH4NNS1
Or via the .msix from the Release page https://github.com/microsoft/winget-cli/releases
Both SHOULD be User installations without Administrator rights (Or does msix require administrator and its appxbundle that does not need administrator?)

However this error
> ResourceUnavailable: Program 'winget.exe' failed to run: An error occurred trying to start process 'C:\Windows\system32\winget.exe' with working directory 'C:'. The file cannot be accessed by the system

could come from a Antivirus blocking the winget.exe from launching.
Imo your IT department could look into the WinGet issue, if they bring up you're not allowed to install anything you could say anything that does not install Machine-Wide you can download from the Web and install without issue. However they might bring up that the Antivirus will block any SHA hash that they do not approve. (which is why winget.exe is blocked)

**Comment by KirkMunroSagent** [comment, marked as answer]
For reference for other folks, I worked this out.

What I had to do:
1. Run Windows PowerShell using the admin account that does not have interactive logon rights, and remove the Microsoft.DesktopAppInstaller and Microsoft.Winget.Source Appx packages.
2. Run Windows PowerShell using user account and remove the same Appx packages.
3. In an elevated PowerShell session, delete C:\Windows\System32\winget.exe.
4. Verify that Microsoft.DesktopAppInstaller and Microsoft.Winget.Source do not show up as Appx packages for any users.
5. Manually install the latest msixbundle from this repo.

Once I did that, winget started working again.
